tools/template_based_query_generation/src/main/java/parser/Pair.java

```diff
@@ -0,0 +1,16 @@
+package parser;
+
+public class Pair<F, S> {
```

Comment:
Java has it's own pair class. Any reason to create another pair class here?
https://docs.oracle.com/javase/8/javafx/api/javafx/util/Pair.html

Reply:
For some reason, Intellij + maven is giving me difficulties with javafx (this was why I decided to do it myself originally). I have deleted the class and used import internal.net.http.common.Pair instead

Reply:
So turns out that package isn't that great to work with either. This one was the one that worked in the end: org.apache.commons.lang3.tuple.MutablePair